documentation/modules/ROOT/pages/development/engine.adoc

```diff
@@ -480,6 +513,44 @@ The default is a periodic commity policy based upon time intervals.
 |Max delay (in ms) between retries when encountering connection errors.
 |===
 
+[[async-engine-properties]]
+=== Asynchronous Engine Properties
+
+[cols="35%a,10%a,55%a"]
+|===
+|Property
+|Default
+|Description
+
+|`record.processing.threads`
+|`#` of availabel cores
+|The number of threads to be used for processing CDC records. The default is number of available machine cores with upper limit of 16 threads.
+If you want to use all available thread without any limitation, use `AVAILABLE_CORES` placeholder.
+
+|`record.processing.shutdown.timeout.ms`
+|1000
+|Maximum time in milliseconds to wait for processing submitted records when task shutdown is called.
+
+|`record.processing.order`
+|`ORDERED`
+|Determines how the records should be produced.
+
+* `ORDERED` means sequential processing, i.e. that the records are produced in the same order as they were obtained from the database.
+* `UNORDERED` means non-sequential processing, i.e. the records can be produced in a different order than the original one.
+
+Non-sequential approach gives better throughput, as the records are produced immediately once the SMTs and serialization of the message is done, without waiting of other records.
```

Comment:
```suggestion
The non-sequential processing of the `UNORDERED` option results in better throughput, because records are produced immediately after any SMT processing and message serialization is complete, without waiting for other records.
```